backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsync.java

```diff
@@ -0,0 +1,168 @@
+        String country =
+            extractedData.getCountry().getValue() == null
+                ? "USA"
+                : extractedData.getCountry().getValue();
+
+        if (_personService.isDuplicatePatient(
+            extractedData.getFirstName().getValue(),
+            extractedData.getLastName().getValue(),
+            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+            currentOrganization,
+            assignedFacility)) {
+          continue;
+        }
+
+        // create new person with current organization, then add to new patients list
+        Person newPatient =
+            new Person(
+                currentOrganization,
+                assignedFacility.orElse(null),
+                null, // lookupid
+                extractedData.getFirstName().getValue(),
+                extractedData.getMiddleName().getValue(),
+                extractedData.getLastName().getValue(),
+                extractedData.getSuffix().getValue(),
+                parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+                address,
+                country,
+                parsePersonRole(extractedData.getRole().getValue(), false),
+                List.of(extractedData.getEmail().getValue()),
+                convertRaceToDatabaseValue(extractedData.getRace().getValue()),
+                convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
+                null, // tribalAffiliation
+                convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
+                parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
+                parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
+                null, // preferredLanguage
+                null // testResultDeliveryPreference
+                );
+
+        if (!patientsList.contains(newPatient)) {
+          // collect phone numbers and associate them with the patient
+          // then add to phone numbers list and set primary phone, if exists
+          List<PhoneNumber> newPhoneNumbers =
+              _personService.assignPhoneNumbersToPatient(
+                  newPatient,
+                  List.of(
+                      new PhoneNumber(
+                          parsePhoneType(extractedData.getPhoneNumberType().getValue()),
+                          extractedData.getPhoneNumber().getValue())));
+          phoneNumbersList.addAll(newPhoneNumbers);
+          newPhoneNumbers.stream().findFirst().ifPresent(newPatient::setPrimaryPhone);
+
+          patientsList.add(newPatient);
+        }
+      } catch (IllegalArgumentException e) {
+        String errorMessage = "Error uploading patient roster";
+        log.error(
+            errorMessage
+                + " for organization "
+                + currentOrganization.getExternalId()
+                + " and facility "
+                + facilityId);
+        throw new IllegalArgumentException(errorMessage);
```

Comment:
Wouldn't we also want to send an email if the upload failed?

Reply:
@BobanL that is an excellent question. Let me check in with design about that.

Reply:
+1 to this, I think we'd want some kind of "try again or contact support" messaging.

Reply:
@BobanL @emmastephenson please take a look at the new changes and let me know if the error handling makes sense to you, specifically in this area:
https://github.com/CDCgov/prime-simplereport/blob/7c2ecfa52a74f6f4845474e4458d54b5f6cd83ff/backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsync.java#L153-L153

Reply:
Note that it's really not possible to force this error in an environment, so I can't really have you test it manually! But once Emma's async unit test is ready, I'll add on to that.